File: front/src/model/ResponseReview.ts
import type { ResponseImage } from './ResponseImage';
import type { ResponseItemReview } from './ResponseItemReview';
import type { Shop } from './Shop';

export interface ResponseReview {
  id: number;
  rating?: number;
  content?: string;
  shop: Shop;
  items?: ResponseItemReview[];
  images?: ResponseImage[];
  createdAt: string;
}

File: src/main/java/com/gk/gourmet_note/review/vo/ResponseReview.java
package com.gk.gourmet_note.review.vo;

import com.gk.gourmet_note.image.vo.ResponseImage;
import com.gk.gourmet_note.shop.vo.ResponseShop;
import lombok.Builder;

import java.time.LocalDateTime;
import java.util.List;

@Builder
public record ResponseReview(
        Long id,
        Float rating,
        String reviews,
        ResponseShop shop,
        LocalDateTime createdAt,
        List<ResponseItemReview> items,
        List<ResponseImage> images
) {
}
